docs/dev-manuals/adr/0002-use-graphql-federation.md

```diff
@@ -0,0 +1,48 @@
+---
+sidebar_position: 3
+---
+
+# Use GraphQL Federation for API Gateway
+
+## Context and Problem Statement
+
+As we decided to use GraphQL for the API gateway, we need a technology which allows us to combine the API of different services into one unified GraphQL API.
+
+## Considered Options
+
+- GraphQL Mesh / Schema stichting
+- GraphQL Federation
+
+## Decision Outcome
+
+Chosen option: "GraphQL Federation", because
+
+- Combining different GraphQL APIs simpler than combinding non-GraphQL APIs (in particular regarding type merging), and GraphQL federation is easier to use in this case
+- Flexibility of GraphQL Mesh not required when using GraphQL Federation compatible GraphQL APIs
+
+### Consequences
+
+- Neutral, because requires us to setup a GraphQL Federation compatiple API gateway (router)
+- Neutral, because requires each service to provide a GraphQL Federation compatible GraphQL API
+
+## Pros and Cons of the Options
+
+### GraphQL Mesh / Schema stichting
+
+GraphQL Mesh is a library allowing to combine multiple data sources into one unified GraphQL API.
+Supported data formats include other GraphQL APIs, REST APIs, gRPC APIs, and more.
+
+- Good, because flexible: allows to combine different data sources
+- Good, because allows to transform API schemas before combining them
+- Neutral, because only provides significant benefits when used with non-GraphQL APIs, but e.g. using REST APIs results in challenges, in particular with respect to combining types
```

Comment:
I don't think we can rely on all routes being GraphQL.
While the **data-fetching** is GraphQL, I am not so sure about data manipulation.
Or do we implement all data manipulation as Event messaging?
If that's the case, document it.

Reply:
GraphQL has mutations, why not use these for data manipulations where applicable?
This only refers to the incoming request, of course, this could cause further non-GraphQL calls internally.
Or did I misunderstand you?